.github/workflows/release-nightly.yml

```diff
@@ -0,0 +1,66 @@
+name: 'Stencil Nightly Release'
+
+on:
+  schedule:
+    # Run every Monday-Friday at 5:00 UTC (5:00 am UTC)
+    # This is done to have a nightly build ready for the Ionic Framework/Stencil Eval Workflow:
+    # https://github.com/ionic-team/ionic-framework/blob/main/.github/workflows/stencil-eval.yml
+    - cron: '00 05 * * 1-5'
+
+jobs:
+  build_core:
+    name: Build
+    uses: ./.github/workflows/build.yml
+
+  get-nightly-version:
+    name: Get Nightly Build Version
+    needs: [build_core]
+    runs-on: ubuntu-20.04
+    outputs:
+      nightly-version: ${{ steps.get-nightly-version.outputs.NIGHTLY_VERSION }}
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+
+      - name: Get Core Dependencies
+        uses: ./.github/workflows/actions/get-core-dependencies
+
+      - name: Download Build Archive
+        uses: ./.github/workflows/actions/download-archive
+        with:
+          name: stencil-core
+          path: .
+          filename: stencil-core-build.zip
+
+      - name: Get Version
+        id: get-nightly-version
+        run: |
+          # A unique string to publish Stencil under, combining the values above
+          # e.g. "3.0.1-dev.1677185104.7c87e34"
+          #
+          # Note: A 'nightly' build is just a 'dev' build that is published at
+          # night, under the 'nightly' tag in npm
+          #
+          # Pull this value from the compiled artifacts
+          NIGHTLY_VERSION=$(./bin/stencil version)
+
+          echo "Using version $NIGHTLY_VERSION"
+
+          # store a key/value pair in GITHUB_OUTPUT
+          # e.g. "NIGHTLY_VERSION=3.0.1-dev.1677185104.7c87e34"
+          echo "NIGHTLY_VERSION=$NIGHTLY_VERSION" >> $GITHUB_OUTPUT
+
+        shell: bash
+
+  release-stencil-nightly-build:
+    name: Publish Nightly Build
+    needs: [get-nightly-version, build_core]
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      - uses: ./.github/workflows/actions/publish-npm
+        with:
+          tag: nightly
+          version: ${{ needs.get-nightly-version.outputs.nightly-version }}
```

Comment:
For my own knowledge, does Github accept the different variable formats? What I mean is:

- In the `get-nightly-version` step, we output the value to a variable `NIGHTLY_VERSION` (uppercase, snake case)
- Here, we grab the output under the variable `nightly-version`

I know you tested this, so it must work, just didn't know actions were so loose with their syntax 😂

Reply:
Yeah, it's a little funky. `NIGHTLY_VERSION` is bash nomenclature for constants + env  variable names.


We then push that to `steps.get-nightly-version.outputs`, giving us `steps.get-nightly-version.outputs.NIGHTLY_VERSION`, which we then reassign effectively to `nightly-version` in https://github.com/ionic-team/stencil/pull/4100/files#diff-e3889abd7e85ba6d5d9dec9274d72a1ce3314fe8169ec90742fecefb40dbe554R20:
```yaml
  get-nightly-version:
    name: Get Nightly Build Version
    needs: [build_core]
    runs-on: ubuntu-20.04
    outputs:
      nightly-version: ${{ steps.get-nightly-version.outputs.NIGHTLY_VERSION }}
```

We _could_ make this a lower cased variable like GH has in their docs https://docs.github.com/en/actions/using-jobs/defining-outputs-for-jobs. I followed [Google's style guide](https://google.github.io/styleguide/shellguide.html#constants-and-environment-variable-names) here, because I liked this:
> Constants and anything exported to the environment should be capitalized.

which felt right given we were exporting `NIGHTLY_VERSION`

Reply:
Ope, I missed that `output`... thanks for bringing that to my attention!